consequence_prediction/vep_mapping_pipeline/consequence_mapping.py

```diff
@@ -69,11 +71,35 @@ def query_vep(variants, search_distance):
     return result.json()
 
 
+@lru_cache
+@retry(tries=10, delay=5, backoff=1.2, jitter=(1, 3), logger=logger)
+def query_consequence_types():
+    url = 'https://rest.ensembl.org/info/variation/consequence_types?content-type=application/json&rank=1'
+    result = requests.get(url)
+    result.raise_for_status()
+    return result.json()
+
+
+def get_so_accessions():
+    consequence_type_results = query_consequence_types()
+    return {conseq['SO_term']: conseq['SO_accession'] for conseq in consequence_type_results}
+
+
+def get_severity_ranking():
+    consequence_type_results = query_consequence_types()
+    # Some terms have the same rank, for these we sort lexicographically within a rank to get a stable ordering.
```

Comment:
To be honest this doesn't sit well with me. If I understood correctly, when VEP states different genes are affected in synchrony, we take the most severe consequence of them all (this issue https://github.com/EBIvariation/eva-opentargets/issues/303#issuecomment-1070981464), and we would be doing it arbitrarily for its naming convention. 

I understand is probably the only thing we can do for now. Besides, as Daniel mentioned, the **consequence is not that relevant for OT** as long as the affected genes are there. Nevertheless, whoever analyses our data should be reminded that this approach will increase the amount of certain consequences and decrease the amount of their siblings in ranking. If we expanded the evidence strings when we found multiple genes/consequences, there wouldn't be an issue in ranking them differently, but picking only the top makes it a key step. 

Have we considered how many consequences in combination we would be arbitrarily fixing? Are Ensembl naming conventions for consequences static between releases?

Reply:
Yeah, I definitely agree with what you're saying here!  I think in the short term this is an acceptable (and like you say, probably only) solution.  In the slightly-less-short term I'd much rather implement something like #321 to mitigate some of these effects.

Ensembl naming conventions for consequences are indeed static and conform to SO terms, and the lexicographic ordering bit is affecting a relatively small number of terms.  Here's the ranking directly from the API:
```
1 ['transcript_ablation']
3 ['splice_acceptor_variant', 'splice_donor_variant']
4 ['stop_gained', 'splice_donor_5th_base_variant']
5 ['frameshift_variant']
6 ['stop_lost']
7 ['start_lost']
8 ['transcript_amplification']
10 ['inframe_insertion']
11 ['inframe_deletion']
12 ['protein_altering_variant', 'missense_variant']
13 ['splice_region_variant', 'splice_polypyrimidine_tract_variant']
14 ['splice_donor_region_variant', 'incomplete_terminal_codon_variant']
15 ['synonymous_variant', 'stop_retained_variant', 'start_retained_variant']
16 ['coding_sequence_variant']
17 ['mature_miRNA_variant']
18 ['5_prime_UTR_variant']
19 ['3_prime_UTR_variant']
20 ['non_coding_transcript_exon_variant']
21 ['intron_variant']
22 ['NMD_transcript_variant']
23 ['non_coding_transcript_variant']
24 ['upstream_gene_variant']
25 ['downstream_gene_variant']
26 ['TFBS_ablation']
28 ['TFBS_amplification']
30 ['TF_binding_site_variant']
31 ['regulatory_region_ablation']
33 ['regulatory_region_amplification']
36 ['feature_elongation', 'regulatory_region_variant']
37 ['feature_truncation']
38 ['intergenic_variant']
39 ['sequence_variant']
```

And here's a diff between the `severity_ranking.txt` file we were using previously for ranking, and the ranking this code generates:

![rank-diff](https://user-images.githubusercontent.com/5599307/164463333-d2e14b42-4aa1-4ac5-b78e-22ef9087b0c0.png)

So currently the only differences are the new terms added in Ensembl 105 (that we guessed the rank for)...  You can also check [Ensembl's documentation](https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html#consequences) which is now finally updated with the new terms.

Hopefully that alleviates some of your concerns, but let me know what you think!

Reply:
I'm surprised the difference between the new and previous rankings is just the new ones 🤔 
Either a big coincidence or we also ranked them lexicographically. In any case, I'm good with the changes and glad it doesn't affect it too much!